### PR TITLE
BGS Find Regional Offices

### PR DESCRIPTION
**Pull request by kathleencrawford**
## Description of change

During integration testing with BGS for the Form 686c (Dependency Application), it was discovered that we need to look up the Location ID of the Regional Office closest to the Veteran and submit that location id as part of the form submission.  This PR adds a new BGS call during the 686 form submission to get and set the regional office location id.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#14974

## Things to know about this PR

This is a pretty minor PR that is just adding to an already existing form submission workflow, so no new settings or swagger docs have been added.  The method added does log a warning to sentry, but no major changes.

- [x] Local testing
- [x] Specs added and updated

### CODE REVIEW

File: lib/bgs/service.rb [hunk 1]
```diff
@@ -196,6 +196,13 @@ def get_regional_office_by_zip_code(zip_code, country, province, lob, ssn)
       '347' # return default location id
     end
 
+    def find_regional_offices
+      find_regional_offices_response = service.data.find_regional_offices
+      find_regional_offices_response[:return]
```
Comment (ericboehs) [inline lib/bgs/service.rb:201]:
```suggestion
      find_regional_offices_response = service.data.find_regional_offices[:return]
```

Would this work?

**Review by ericboehs — APPROVED** [review]
It might be good to test the other code paths for location_id (I see one of the three is tested).

Approved nonetheless. 👍

File: lib/bgs/service.rb (hunk 1 above)
Comment (kathleencrawford) [reply lib/bgs/service.rb]: Yes!  Thank you - I've updated as suggested.

**Comment by kathleencrawford** [comment]
> It might be good to test the other code paths for location_id (I see one of the three is tested).
> 
> Approved nonetheless. 👍

@ericboehs I've added 2 new tests as suggested.  Thank you for your review and suggestions!

**Pull request merged**